public/app/plugins/datasource/loki/tracking.ts

```diff
@@ -0,0 +1,23 @@
+import { DataQueryResponse } from '@grafana/data';
+import { reportInteraction } from '@grafana/runtime';
+
+import { isLogsQuery, parseToArray } from './queryUtils';
+import { LokiQuery } from './types';
+
+export function trackQuery(response: DataQueryResponse, queries: LokiQuery[], app: string): void {
+  for (const query of queries) {
+    reportInteraction('grafana_loki_query_executed', {
+      app,
+      editor_mode: query.editorMode,
+      has_data: response.data.some((frame) => frame.length > 0),
+      has_error: response.error !== undefined,
+      legend: query.legendFormat,
+      line_limit: query.maxLines,
+      parsed_query: parseToArray(query.expr),
+      query_type: isLogsQuery(query.expr) ? 'logs' : 'metrics',
```

Comment:
I know we use `metrics query` at couple of places, but it is supposed to be `metric query` based on https://grafana.com/docs/loki/latest/logql/metric_queries/. We should probably use `metric` and in the future probably create enum to make this uniform. 
```suggestion
      query_type: isLogsQuery(query.expr) ? 'logs' : 'metric',
```